src/time/format/parser.cr

```diff
@@ -166,14 +166,27 @@ struct Time::Format
     def milliseconds
       # Consume more than 3 digits (12 seems a good maximum),
       # and later just use the first 3 digits because Time
-      # only has microsecond precision.
+      # need millisecond precision.
       pos = @reader.pos
-      millisecond = consume_number(12)
+      millisecond = consume_number_i64(12)
```

Comment:
I don't think there is a need for 64 bit integer here. You should use `consume_number`.

Reply:
1) this value will be multiplied with `Int64` later. Less conversations needed.
2) `consume_number` is just `consume_number_i64.to_i`.
3) parsing of 12 numbers at once is just current reasonable choice (picoseconds). It may be changed later.